apps/highlight/src/agents/townhall.ts

```diff
@@ -158,4 +160,22 @@ export default class Townhall extends Agent {
   async deleteRole({ space, id }: { space: string; id: string }) {
     this.emit('delete_role', [space, id]);
   }
+
+  async claimRole(
+    { space, id }: { space: string; id: string },
+    { signer }: { signer: string }
+  ) {
+    const user = await this.getAuthor(signer);
+
+    this.emit('claim_role', [space, id, user]);
+  }
+
+  async revokeRole(
+    { space, id }: { space: string; id: string },
+    { signer }: { signer: string }
+  ) {
+    const user = await this.getAuthor(signer);
```

Comment:
The function name `getAuthor` is misleading now that's it's used beyond posting discussion, it's more `getSigner`